mpmath/__init__.py

```diff
@@ -432,6 +436,34 @@
 unit_triangle = mp.unit_triangle
 sigmoid = mp.sigmoid
 
+
+class SetMPMathPrecisionError(RuntimeError):
```

Comment:
AttributeError?

Reply:
Here is a version without this class (nothing goes to the global mpmath namespace):
```diff
diff --git a/mpmath/__init__.py b/mpmath/__init__.py
index 46a7c6f..6fcaaae 100644
--- a/mpmath/__init__.py
+++ b/mpmath/__init__.py
@@ -1,5 +1,9 @@
 __version__ = '1.3.0'
 
+import functools
+import sys
+import types
+
 from .usertools import monitor, timing
 
 from .ctx_fp import FPContext
@@ -432,6 +436,20 @@
 unit_triangle = mp.unit_triangle
 sigmoid = mp.sigmoid
 
+
+class _MPMathModuleType(types.ModuleType):
+
+    def _helper(self, *args, prop=''):
+        raise AttributeError("cannot set '{name}' on 'mpmath'. Did you mean to "
+                             "set '{name}' on 'mpmath.mp'?".format(name=prop))
+
+    dps = property(fset=functools.partial(_helper, prop='dps'))
+    prec = property(fset=functools.partial(_helper, prop='prec'))
+
+
+sys.modules[__name__].__class__ = _MPMathModuleType
+del functools, types, sys
+
 # be careful when changing this name, don't use test*!
 def runtests():
     """
```

Reply:
yes, I like this better too